src/App.tsx

```diff
@@ -1,27 +1,65 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import './App.scss';
 import { GoodsList } from './GoodsList';
+import { getAll, get5First, getRedGoods } from './api/goods';
+import { Good } from './types/Good';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+export const App: React.FC = React.memo(() => {
+  const [goods, setGoods] = useState<Good[]>([]);
+  const [selected5First, setSelected5First] = useState(false);
+  const [selectedRed, setSelectedRed] = useState(false);
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+  useEffect(() => {
+    if (selected5First) {
+      get5First()
+        .then(setGoods);
+    } else if (selectedRed) {
+      getRedGoods()
+        .then(setGoods);
+    } else {
+      getAll()
+        .then(setGoods);
+    }
+  }, [goods]);
```

Comment:
Redundant here. Look, you can creat functions for each button and wrapped it on useCallback as Anastasia note in previous review. Or alternative You can create memoize value to prevent make request each time when you click on button, for example:
`const allTodos = useMemo(() => getAll(), []) const firstFiveTodos= useMemo(() => get5First(), [])`
The same for first 5 goods and red goods. And after that you can set this memoized values to state by clicking corresponding button